src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
I suggest `ticklabelorder` or just `tickorder`. It is in the family of:
- `showticklabels`
- `tickprefix`, `ticksuffix`
- `tickfont`
```

For values, I suggest:
- I find `trace` ambiguous. How about `"ticklabels"`? Alternatively `data`, `default`, `''`, ...
- `category ascending` -> `ascending`: personally, I think it's clear that ascending refers to `A`->`Z`. This language is used in a lot other places: SQL (`SORT BY X ASC`), [Excel](http://content.gcflearnfree.org/topics/177/07_02_02_SortCommand.png). This should work across dates, numbers, and strings (without the user needing to specify different values depending on the axis type)
- `array` -> `categorylist`: more explicit about where you go about to actually set the values used to sort.


We'll also want eventually need to add:
- `case-insensitive-ascending`
- `case-insensitive-descending`

> there was discussion on the issue about eventually adding value-based sort options, e.g. value ascending

If we add this, then we can support it with `categorylist-ascending` and `categorylist-descending` and `categorylist-case-insensitive-ascending` and `categorylist-case-insensitive-descending`

Related, how does this relate to the `"ticklabels"` property?